src/test/codegen/likely.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// compile-flags: -C no-prepopulate-passes
+
+#![crate_type = "lib"]
+#![feature(core_intrinsics)]
+
+use std::intrinsics::{likely,unlikely};
+
+#[no_mangle]
+pub fn check_likely(x: i32, y: i32) -> Option<i32> {
+    unsafe {
+        // CHECK: call i1 @llvm.expect.i1(i1 %{{.*}}, i1 true)
```

Comment:
It seems there's a couple of casts and things in between. This is the relevant part of the `likely.ll` file:

```
start:
  %2 = icmp eq i32 %0, %1
  %3 = call i1 @llvm.expect.i1(i1 %2, i1 true)
  %4 = zext i1 %3 to i8
  store i8 %4, i8* %tmp_ret
  %5 = load i8, i8* %tmp_ret, !range !0
  %6 = trunc i8 %5 to i1
  br label %bb1

bb1:
  br i1 %6, label %bb2, label %bb3
```